### ISSUE TITLE
Unable to sign in

### ISSUE THREAD

**Issue by wish2snowboard** [eshapiro42/OctoPrint-Wyze#11]
Python version 3.9.2, Octoprint 1.8.6, OctoPi 1.0.0

Cannot sign into Wyze account. Installed the plugin, went to OctoPrint-Wyze settings, entered credentials, hit save, restart OctoPrint. When I go back into settings the credentials fields are empty. I've tested my credentials through the Wyze app and know they are accurate. 
I tried rebooting the system and uninstalling and cleaning data then reinstalling without any difference. 
The OctoPrint-Wyze tab is present and shows event handler information and the column headers but nothing under them and the top shows "Connected to Wyze as None."

Any advice? Thanks.

**Comment by tbnorris** [comment]
I am not sure if this will help you, but I had a similar problem because I had Two Factor turn on with my Wyze account.  I played with it for a few minutes and then since I had JUST made the Wzye account I decided to see if that was what it was and it instantly logged in at that point.

**Comment by wish2snowboard** [comment]
> I am not sure if this will help you, but I had a similar problem because I had Two Factor turn on with my Wyze account. I played with it for a few minutes and then since I had JUST made the Wzye account I decided to see if that was what it was and it instantly logged in at that point.

Thanks, unfortunately for me multi factor was off to begin with so that doesn't seem to be the issue.

**Comment by eshapiro42** [comment]
Hi @wish2snowboard! Sorry you're having this problem and that I've been slow to respond.

Could you please send any console errors logged in your browser's development tools when you try to open the OctoPrint-Wyze tab? As well as any OctoPrint-Wyze-related errors that may pop up in your server logs when you launch the OctoPrint server?

This sort of thing is very difficult to debug since I don't have access to your system but that would be a good starting point.

**Comment by wish2snowboard** [comment]
> Hi @wish2snowboard! Sorry you're having this problem and that I've been slow to respond.
> 
> Could you please send any console errors logged in your browser's development tools when you try to open the OctoPrint-Wyze tab? As well as any OctoPrint-Wyze-related errors that may pop up in your server logs when you launch the OctoPrint server?
> 
> This sort of thing is very difficult to debug since I don't have access to your system but that would be a good starting point.

browser development tools show "Failed to load resource: the server responded with a status of 500 ()" /api/plugin/wyze:1
I've attached my OctoPrint log, picking out the info that may be useful to you is a bit beyond my level.
Thanks for taking a look at this! 

[octoprint.log](https://github.com/eshapiro42/OctoPrint-Wyze/files/10961239/octoprint.log)

**Comment by eshapiro42** [comment]
Thanks, that's very helpful. I'll try to take a look at this soon. The root cause is something on your system preventing the `wyze` object to not get initialized, so I need to narrow down what could be causing that.

```
2023-03-13 15:12:03,830 - octoprint.server.api - ERROR - Error while executing SimpleApiPlugin wyze
Traceback (most recent call last):
  File "/home/pi/oprint/lib/python3.9/site-packages/octoprint/server/api/__init__.py", line 158, in pluginCommand
    response = api_plugin.on_api_command(command, data)
  File "/home/pi/oprint/lib/python3.9/site-packages/octoprint/util/__init__.py", line 1688, in wrapper
    return f(*args, **kwargs)
  File "/home/pi/oprint/lib/python3.9/site-packages/octoprint_wyze/__init__.py", line 128, in on_api_command
    devices = self.wyze.get_devices(self.event_handler)
AttributeError: 'WyzePlugin' object has no attribute 'wyze'
```

**Comment by wish2snowboard** [comment]
> Thanks, that's very helpful. I'll try to take a look at this soon. The root cause is something on your system preventing the `wyze` object to not get initialized, so I need to narrow down what could be causing that.
> 
> ```
> 2023-03-13 15:12:03,830 - octoprint.server.api - ERROR - Error while executing SimpleApiPlugin wyze
> Traceback (most recent call last):
>   File "/home/pi/oprint/lib/python3.9/site-packages/octoprint/server/api/__init__.py", line 158, in pluginCommand
>     response = api_plugin.on_api_command(command, data)
>   File "/home/pi/oprint/lib/python3.9/site-packages/octoprint/util/__init__.py", line 1688, in wrapper
>     return f(*args, **kwargs)
>   File "/home/pi/oprint/lib/python3.9/site-packages/octoprint_wyze/__init__.py", line 128, in on_api_command
>     devices = self.wyze.get_devices(self.event_handler)
> AttributeError: 'WyzePlugin' object has no attribute 'wyze'
> ```

Thank you, let me know if there's anything else I can do.

**Comment by eshapiro42** [comment]
Could you please look at your `.octoprint/config.yaml` file and confirm that it contains a `wyze` section with entries for `wyze_email`, `wyze_key` and `wyze_password`?

If they are present, please simply indicate this and **DO NOT** upload or paste these values as they could easily be decrypted by anyone who views them to reveal your Wyze credentials.

I suspect that these settings are not being saved properly for whatever reason and this is what's preventing initialization of the `wyze` object in my code.

**Comment by wish2snowboard** [comment]
Only mentions of wyze in my config.yaml is

appearance:
  color: black
  components:
    order:
      tab:
      - temperature
      - control
      - plugin_prettygcode
      - plugin_bedlevelvisualizer
      - timelapse
      - plugin_gcodeviewer
      - terminal
      - plugin_wyze
  name: Ender S1

and

tabs: 
- icon: ''
      icon_color: ''
      icon_tooltip: ''
      name: plugin_wyze
      showtext: true
      usetitle: false

**Comment by eshapiro42** [comment]
Interesting, so it seems like the settings didn't get saved after all. Could you try inputting them again in the settings page for the plugin? If they still aren't appearing in your `config.yaml` after that, maybe try uninstalling and reinstalling the plugin. I'm not really sure what could be preventing them from saving.

**Comment by wish2snowboard** [comment]
Still nothing, I've reinstalled somewhere around 5 times. 
I'm seeing this error in the browser dev tools when trying to save settings. 
<img width="1251" alt="Screenshot 2023-03-17 at 5 45 23 PM" src="https://user-images.githubusercontent.com/85231428/226059045-02945bbb-fa1b-42be-ae4e-c2c644a33c84.png">

**Comment by wish2snowboard** [comment]
Just tried a fresh install of OctoPi on a different microSD, figured if it worked I could just rebuild. Same issue as before though. OctoPrint 1.8.7, Python 3.9.2, OctoPi 1.0.0. Screenshot & the configuredloggers has not changed error
![Screenshot 2023-03-19 at 9 31 47 PM](https://user-images.githubusercontent.com/85231428/226226904-291b4e41-f180-473a-9b3b-08a67cec0eae.png)

**Comment by slanthie** [comment]
I am having same issue. I have installed latest Octopi (OctoPrint Version 1.9.3 / OctoPi Build 2023.10.09.151442 ) and installed octoprint-wyze 0.2.1 from the plugin-manager, and when I save username/pwd, it doesn't seem to save. 

From the logs, looks like it's not saving to the config:

```
2023-10-10 20:46:23,166 - octoprint.server.api.settings - ERROR - Could not save settings for plugin OctoPrint-Wyze (0.2.1)
Traceback (most recent call last):
<omitted>
wyze_sdk.errors.WyzeApiError: Parameters passed to Wyze Service do not fit the endpoint
The server responded with: {'ts': 1696967183161, 'code': '1001', 'msg': 'ParameterError', 'data': {}}
```

I checked config.yaml, and I don't see a wyze section there. Is there a way to populate the config.yaml manually?

**Comment by eshapiro42** [comment]
I believe this has been fixed as of release [v0.2.2](https://github.com/eshapiro42/OctoPrint-Wyze/releases/tag/v0.2.2) and was the result of Wyze requiring an API key for authentication.
